### DISCUSSION TITLE
How to automatically connect to the nearest server region with Cloud Functions httpsCallable?

### DISCUSSION THREAD

**Discussion by cemck** [invertase/react-native-firebase#8190] [Q&A]
How to automatically connect to the nearest server region with Cloud Functions httpsCallable? 
I want to reduce latency and use multiple cloud function server regions, is that possible without using device location APIs?

Thanks in advance

**Comment by mikehardy** [comment]
There is no ability to do that in the firebase APIs, and this module is intended to expose the native firebase SDK APIs under a firebase-js-sdk API compatible surface area, nothing more nothing less

What you want is device geolocation, or geocoding to resolve to a location, then a lookup between the functions regions. That would be custom code involving something like react-native-background-geolocation-android and a lookup table of cloud function regions and the lat/long geofences you'd want to go to each one, or some cloud geocoding lookup service in the cloud based on IP and still a lookup table you maintain of cloud function regions and the lat/long you want to go to each one

None of that will happen in this library, it would be an app feature you'd need to implement. This library can support specifying the region for a cloud function call, as permitted in the native SDKs but nothing more - determining the region is out of scope

**Reply by cemck** [reply]
Thank you for the response and for the great library!
